Update dependencies to match gulp 4

doowb opened the pull request:
This PR is to update dependencies to match some that are used in gulp 4.

- removes `gulp-util`
- uses `fancy-log` for DEBUG logging
- updates `findup-sync`
- uses `micromatch` (used in `findup-sync` and `chokidar` for `gulp.watch`)

Since this is an attempt at consolidating common dependencies, I removed the green coloring around the logging messages. If you'd like to keep that, I can add something for it.

callumacrae commented:
Maybe worth using [gulplog](https://www.npmjs.com/package/gulplog)?

Nice work, though :)

jackfranklin commented:
Agree with @callumacrae that I'd use `gulplog` but once that's done awesome work, thank you!

Can you add yourself as a contributor in the `package.json` too please? :)

tunnckoCore commented:
I think it worth using https://github.com/tunnckoCore/load-deps under the hoods, it's pretty fast. Ot just take a look and say some feedback :)

Cheers,
Charlike!

jackfranklin commented:
@tunnckoCore can you open a new issue? It's not related to this PR.

doowb commented:
I updated to use `gulplog` but there are a few caveats:

- gulplog only emits an event so it requires that users are using `gulp-cli` that has the event handling configured.
- `gulp-cli` logs events based on the `-L` flag level so I set the logger to do `logger.debug` which requires the `-LLLL` to be used.
- Changed the `DEBUG` option to default to `true` when undefined. This will still only log when `-LLLL` is used but it allows users to disable it through the `DEBUG` option by passing in `{ DEBUG: false }`.

For the first bullet, we could listen for the events and write to the console but this will result in duplicate logging when users are using `gulp-cli`.

Also, added my name to the contributors list, thanks!

jackfranklin commented on README.md:
```diff
@@ -58,15 +58,15 @@ You can pass in an object of options that are shown below: (the values for the k
     rename: {}, // a mapping of plugins to rename
-    renameFn: function (name) { ... } // a function to handle the renaming of plugins (the default works)
+    renameFn: function (name) { ... }, // a function to handle the renaming of plugins (the default works)
+    DEBUG: true // when set to true and using gulp-cli flag -LLLL, the plugin will log info to console. Useful for bug reporting and issue debugging
```
can we keep this as `false`, as this shows the default values and `DEBUG` should be `false` by default.

jackfranklin commented on index.js:
```diff
@@ -21,7 +22,7 @@ module.exports = function(options) {
   options = options || {};
 
-  var DEBUG = options.DEBUG || false;
+  var DEBUG = typeof options.DEBUG === 'undefined' ? true : options.DEBUG;
```
`DEBUG` should be `false` by default.

doowb commented:
@jackfranklin I reverted the DEBUG changes per your inline comments.

jackfranklin commented:
Thank you! @callumacrae does this look good to you? :)

callumacrae commented:
Yep, looks good!

doowb commented:
@jackfranklin thanks for merging. Will you publish to npm?

jackfranklin commented:
@doowb released as 1.2.3. Thanks :)
